packages/cli/src/environments/versionControl/versionControl.service.ee.ts

```diff
@@ -0,0 +1,75 @@
+import { Service } from 'typedi';
+import { generateSshKeyPair } from './versionControlHelper';
+import { VersionControlPreferences } from './types/versionControlPreferences';
+import { VERSION_CONTROL_PREFERENCES_DB_KEY } from './constants';
+import * as Db from '@/Db';
+import { jsonParse } from 'n8n-workflow';
+import type { Settings } from '@/databases/entities/Settings';
+
+@Service()
+export class VersionControlService {
+	private _versionControlPreferences: VersionControlPreferences = new VersionControlPreferences();
+
+	async init(): Promise<void> {
+		await this.loadFromDbAndApplyVersionControlPreferences();
+	}
+
+	public get versionControlPreferences(): VersionControlPreferences {
+		return {
+			...this._versionControlPreferences,
+			privateKey: '',
+		};
+	}
+
+	async generateAndSaveKeyPair(keyType: 'ed25519' | 'rsa' = 'ed25519') {
```

Comment:
I'm guessing this function could check if a key already exists and throw an error in case it does - or at least in the future, check if there is a connected repo.

If a repo is connected, we want to prevent generating new keys at random WDYT?